test/api/notifications_test.dart

```diff
@@ -210,18 +224,29 @@ void main() {
       check(parse({ ...baseJson, 'awesome_feature': 'enabled' })).jsonEquals(baseline);
     });
 
+    test('uses deprecated fields when newer fields are missing', () {
+      final baseline = parse(baseJson);
+
+      // FL 257 deprecated 'realm_uri' in favor of 'realm_url'.
+      final jsonSansRealm =
+        { ...baseJson }..remove('realm_url')..remove('realm_uri');
+      check(parse({ ...jsonSansRealm, 'realm_url': 'https://zulip.example.com/' })).jsonEquals(baseline);
```

Comment:
bump; and I think in fact this revision took this point in the opposite direction :slightly_smiling_face: 

I believe the point of this test is to check that when the old name is present, and not the new name, we get the data from the old name. So the input this test constructs to pass to `parse` should use the old name.

Reply:
Right, sorry about that! Updated to use the older field.